autogen/oai/client.py

```diff
@@ -107,18 +314,23 @@ def __init__(self, *, config_list: Optional[List[Dict[str, Any]]] = None, **base
         openai_config, extra_kwargs = self._separate_openai_config(base_config)
         if type(config_list) is list and len(config_list) == 0:
             logger.warning("openai client was provided with an empty config_list, which may not be intended.")
+
+        self._clients: List[Client] = []
+        self._config_list: List[Dict[str, Any]] = []
+
         if config_list:
             config_list = [config.copy() for config in config_list]  # make a copy before modifying
-            self._clients: List[OpenAI] = [
-                self._client(config, openai_config) for config in config_list
-            ]  # could modify the config
-            self._config_list = [
-                {**extra_kwargs, **{k: v for k, v in config.items() if k not in self.openai_kwargs}}
-                for config in config_list
-            ]
+            for config in config_list:
+                activated = self._register_openai_client(config, openai_config)  # could modify the config
+                self._config_list.append(
+                    {
+                        "config": {**extra_kwargs, **{k: v for k, v in config.items() if k not in self.openai_kwargs}},
+                        "activated": activated,
+                    }
+                )
         else:
-            self._clients = [self._client(extra_kwargs, openai_config)]
-            self._config_list = [extra_kwargs]
+            activated = self._register_openai_client(extra_kwargs, openai_config)
```

Comment:
Here we may want to emit a warning because we are getting an empty config list and assuming we can get everything from the environment variables. If there is no key in the environment variables, there is no error here. But when `create` is called, they get error from OpenAI saying something about `api_key` but no instruction to fix it. 

We can warn the user here first, saying empty config list is received and will default to use environment variables for OpenAI and Azure OpenAI models only.